Examples/BinaryBH/params.txt

```diff
@@ -187,4 +187,41 @@ modes = 2 0 # l m for spherical harmonics
 # extraction_file_prefix = "Weyl4_extraction_"
 
 #################################################
+# Apparent Horizon Finder parameters
+
+AH_activate = 1
+AH_num_ranks = 65
```

Comment:
I managed to segfault the code by using fewer ranks (64) than this number - is there a way to check this at runtime and Mayday if set too high?

Reply:
I think if we can't check this at runtime, then we should reduce this to 64 ranks, as it is very possible that someone will choose to run it on 64 ranks initially (as there are <64 boxes on all levels), and then have issues later when more boxes are created. But maybe @mirenradia  can implement a check.

Reply:
I'm confused by your first comment. You say you got a segfault when you used *fewer* ranks but then you want to check if it is set *higher*? 

Tbh, PETSc decides whether or not it is happy with the number of ranks and I can't find anything concrete in the documentation about how it does this. From reading [this PETSc manual page](https://petsc.org/release/manual/vec/), I think `AH_num_ranks` must divide `AH_num_points_u * AH_num_points_v` which we could try and check for. Having said that, this might be only necessary but not sufficient as I vaguely recall getting errors with values that do satisfy this condition.

Reply:
I want to check if the AH number of ranks is set higher than the number I actually have at runtime. I would have thought it happens automatically but apparently not... admittedly the segfault I got could have been caused by something else. I don't mind just leaving this.

Reply:
Oh as in the number of AH ranks being larger than the total number of ranks for the whole simulation? Yes, I think checking for that would be a good idea.

Should I also add a check for `AH_num_ranks % (AH_num_points_u * AH_num_points_v) == 0` or should we just leave this up to PETSc?

Reply:
Maybe add it as it might help someone - it is not too much of a restriction and having an error message will make it easier for people to debug

Reply:
Ok, I've made these changes.